helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/SoftConstraint.java

```diff
@@ -86,11 +60,11 @@ float getAssignmentNormalizedScore(AssignableNode node, AssignableReplica replic
   }
 
   /**
-   * The default scaler function that squashes any score within (min_score, max_score) to (0, 1);
+   * The default normalize function f(x) = tanh(1/(\alpha * x)) that squashes any score >= 0 to (0, 1]
+   * The hyper parameter determines the converging rate of scores to 0 when X gets larger
    * Child class could override the method and customize the method on its own
-   * @return The MinMaxScaler instance by default
    */
   NormalizeFunction getNormalizeFunction() {
-    return (score) -> (score - getMinScore()) / (getMaxScore() - getMinScore());
+    return score -> (score == 0f ? 1f : (float) Math.tanh(1 / score));
```

Comment:
I tried this formula a little bit. So the range is (0, 1], as expected. One concern is that while I try Math.tanh((float)20), the value returned by Math lib is 1. That means any utilization between 0% and 5% would be evaluated the same by this implementation, right?
Can we make it more accurate, at least to 1% or even 0.1%. I think we can setup the alpha for this, right?

Reply:
I see. You'd like to have a more distinguished value distribution from 1% - 100%. I could leave the alpha and let each soft constraint that's focused on **usage percentage** to have a default alpha of something.

Reply:
Yeah, my feeling is that the constraint should be more sensitive internally so even one partition assignment would cause a difference. Actually, let's ensure the score would be different when there is more than 0.1% usage. And please comment on this in the code so if we want to adjust alpha later, we will have a clue.
Thanks!

Reply:
A simple alpha = 1000 is able to have a sensitivity of 0.1%. In practice, do we have need 0.1% sensitivity or 1% is fine?

Reply:
Let do 0.1%. I personally feel safer with that as a start point : )